urllib3/fields.py

```diff
@@ -36,11 +36,11 @@ def format_header_param(name, value):
         result = '%s="%s"' % (name, value)
         try:
             result.encode('ascii')
-        except UnicodeEncodeError:
+        except (UnicodeEncodeError, UnicodeDecodeError):
             pass
         else:
             return result
-    if not six.PY3:  # Python 2:
+    if not six.PY3 and isinstance(value, unicode):  # Python 2:
```

Comment:
I took a look at six; the issue there is that six.text_type evaluates as true for Python 3 strings as well as Python 2 unicode, and encode_rfc2231 doesn't take Python 3 bytes.